raiden/tests/mocked/test_matrix_transport.py

```diff
@@ -161,13 +197,70 @@ def test_sending_nonstring_body(  # pylint: disable=unused-argument
 def test_processing_invalid_message_json(  # pylint: disable=unused-argument
     mock_matrix, skip_userid_validation, message_input
 ):
-    room, event = make_message(overwrite_data=message_input)
+    room, event = make_message_text(overwrite_data=message_input)
     assert not mock_matrix._handle_sync_messages([(room, [event])])
 
 
 def test_processing_invalid_message_type_json(  # pylint: disable=unused-argument
     mock_matrix, skip_userid_validation
 ):
     invalid_message = '{"_type": "NonExistentMessage", "is": 3, "not_valid": 5}'
-    room, event = make_message(overwrite_data=invalid_message)
+    room, event = make_message_text(overwrite_data=invalid_message)
     assert not mock_matrix._handle_sync_messages([(room, [event])])
+
+
+@pytest.mark.parametrize("retry_interval", [0.01])
+def test_retry_queue_send_only_once(mock_matrix, record_sent_messages):
+    # Pretend the Transport greenlet is running
+    mock_matrix.greenlet = True
+
+    # This is intentionally not using ``MatrixTransport._get_retrier()`` since we don't want the
+    # greenlet to run but instead manually call its `_check_and_send()` method.
+    retry_queue = _RetryQueue(transport=mock_matrix, receiver=factories.HOP1)
+
+    message = make_message()
+    serialized_message = MessageSerializer.serialize(message)
+    queue_identifier = QueueIdentifier(
+        recipient=factories.HOP1, canonical_identifier=CANONICAL_IDENTIFIER_UNORDERED_QUEUE
+    )
+    retry_queue.enqueue(queue_identifier, message)
+
+    mock_matrix._queueids_to_queues[queue_identifier] = [message]
+
+    with retry_queue._lock:
+        retry_queue._check_and_send()
+
+    assert len(mock_matrix.sent_messages) == 1
+    assert (factories.HOP1, serialized_message) in mock_matrix.sent_messages
+
+    mock_matrix._queueids_to_queues[queue_identifier].clear()
+
+    # The retry interval for this test is set to 0.01 (see parametrize). Make sure we're past this.
+    gevent.sleep(0.05)
```

Comment:
please use the fixture here, it will prevent errors if in the future we just change the fixture without reading all the tests.